Maple2.Server.Game/Model/Field/Actor/ActorState/AiState.cs

```diff
@@ -0,0 +1,447 @@
+﻿using Maple2.Model.Enum;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Model.Skill;
+using Maple2.Server.Game.Packets;
+using System.Diagnostics.CodeAnalysis;
+using System.Numerics;
+using static Maple2.Model.Metadata.AiMetadata;
+
+namespace Maple2.Server.Game.Model.Field.Actor.ActorState;
+
+public class AiState {
+    private readonly FieldNpc actor;
+    public AiMetadata? AiMetadata { get; private set; }
+    private Node? battle = null;
+
+    private List<StackEntry> aiStack = new List<StackEntry>();
+
+    private long nextUpdate = 0;
+    private AiMetadata? lastEvaluated = null;
```

Comment:
this and `battle`, we don't need to initialize them with these values as they are _the_ the default values

Reply:
`battle` was made specifically so that it would fit in the stack without muddying the interface too much. I could go and make the change to AiMetadata instead for that compatibility, but I'm also worried about the semantic implications of making that change & how it visually changes the meaning of what it is on AiMetadata. You think I should move it there?

For lastEvaluated that was for checking if the stack was stale in the case that the AiMetadata was swapped & not made null. Stacks aren't compatible across AiMetadatas. I just forgot to add the write & check for it

Reply:
what I mean is
`private Node? battle = null;` is equal to `private Node? battle;`
A tiny detail but just wanna keep the same style as much as possible.

Reply:
Ahh gotcha. Yeah I'll make that change